src/hooks/useDebounce.js

```diff
@@ -0,0 +1,25 @@
+import {useEffect, useRef} from 'react';
+import lodashDebounce from 'lodash/debounce';
+
+/**
+ * @param {Function} func The function to debounce.
+ * @param {Number} wait The number of milliseconds to delay.
+ * @param {Object} options The options object.
+ * @param {Boolean} options.leading Specify invoking on the leading edge of the timeout.
+ * @param {Number} options.maxWait The maximum time func is allowed to be delayed before it’s invoked.
+ * @param {Boolean} options.trailing Specify invoking on the trailing edge of the timeout.
+ * @returns Returns the new debounced function.
+ */
+export default function useDebounce(func, wait, options) {
+    const debouncedFnRef = useRef();
+
+    useEffect(() => {
+        const debouncedFn = lodashDebounce(func, wait, options);
+
+        debouncedFnRef.current = debouncedFn;
+
+        return debouncedFn.cancel;
+    }, [func, wait, options]);
+
+    return debouncedFnRef.current;
```

Comment:
Actually, maybe let's wrap this into a lambda that calls `debouncedFnRef.current` only if it's not `undefined`? I think such cases are always an error, as we don't expect the returned function to be called inside a render function directly, but maybe dropping a code in a corner case is better than a crash?...

Reply:
I know what you mean. Perfectly, it would be great to combine both things, i.e. a safe handling of a corner case + an analytics report of an error. But we don't have such conventions in Expensify.

If you disagree with a suggestion, you can always say so before applying the change. Sometimes C+s will push for a change, sometimes not.